woocommerce/sensei-certificates
# Refactor to load most of plugin after `plugins_loaded`

This is a large refactor to make it possible to load most of the plugin on `plugins_loaded` action hook without breaking activation steps. To do this, it moves most functions out of the main plugin file into the main class and into a new `sensei-certificates-functions.php` file. 

Built file: 
[sensei-certificates.zip](https://github.com/woocommerce/sensei-certificates/files/2979480/sensei-certificates.zip)


### Testing Instructions
With how widespread this PR is, it will require full testing of the plugin. To test the changes in the PR:
- In fresh instance, activate just this plugin (in this branch) without Sensei.
- Activate Sensei and refresh page (see known issues).
- Make sure certificate template is created.

In another fresh instance:
- Activate Sensei. Create courses and lessons. Have multiple users finish those courses.
- Activate this branch of this plugin.
- Make sure certificate template is created and certificates are added to those who completed the course.

### Known Issues
- Without this PR (https://github.com/Automattic/sensei/pull/2490), data updates won't be listed.
- Certificates menu will only load on page load after Sensei or the plugin is activated.
